erpnext/accounts/doctype/purchase_invoice/purchase_invoice.js

```diff
@@ -343,6 +343,9 @@ cur_frm.cscript.select_print_heading = function(doc,cdt,cdn){
 		cur_frm.pformat.print_heading = __("Purchase Invoice");
 }
 
+cur_frm.add_fetch('item_code','weight_per_unit','weight_per_unit');
+cur_frm.add_fetch('item_code','weight_uom','weight_uom');
```

Comment:
Wrote same code in every file. Add this code into transaction.js and check child table(sales order item) has field weight_per_unit